pkg/aws/eni/node.go

```diff
@@ -808,7 +808,11 @@ func (n *Node) GetUsedIPWithPrefixes() int {
 		var pfx string
 		// Check if the IP is from any of the prefixes attached to this ENI
 		for _, ipNet := range prefixNetworks {
-			if ipNet.Contains(net.ParseIP(ip)) {
+			addr, err := netip.ParseAddr(ip)
+			if err != nil {
+				continue
+			}
+			if ipNet.Contains(addr) {
 				prefixBased = true
 				pfx = ipNet.String()
```

Comment:
Instead of converting this to a `string` for map access in `usedPfx` we could now change `usedPfx` to type `m̀ap[netip.Prefix]bool`